src/core/dbus-manager.c

```diff
@@ -1413,17 +1413,47 @@ static int dump_impl(
 
         assert(message);
 
-        /* Anyone can call this method */
-
+        /* 'status' access is the bare minimum always needed for this, as the policy might straight out
+         * forbid a client from querying any information from systemd, regardless of any rate limiting. */
         r = mac_selinux_access_check(message, "status", error);
         if (r < 0)
                 return r;
 
+        /* Rate limit reached? Check if the caller is privileged/allowed by policy to bypass this. We
+         * check the rate limit first to avoid the expensive roundtrip to polkit when not needed. */
+        if (!ratelimit_below(&m->dump_ratelimit)) {
+                /* We need a way for the SELinux policy to allow bypassing rate limits, but we cannot easily
```

Comment:
> We need a way for the SELinux policy to allow bypassing rate limits

That sounds wrong. MAC should only prevent things, never allow something that wouldn't be allowed otherwise. And this check doesn't actually bypass the rate limit, no?

"We need a way for SELinux to constrain the operation when the rate limit is active, even if polkit would allow it. …" ?

Reply:
reworded as suggested